src/connection_string.ts

```diff
@@ -631,7 +652,7 @@ export const OPTIONS = {
     target: 'credentials',
     transform({ options, values: [value] }): MongoCredentials {
       if (typeof value === 'string') {
-        value = toRecord(value);
+        value = Object.fromEntries(parseAuthMechanismParameters(value));
```

Comment:
In order to maintain the previous behavior we can do this, or reevaluate the need

```js
        value = Object.assign(Object.create(null), Object.fromEntries(parseAuthMechanismParameters(value)));
```